presto-feature-toggle/src/test/java/com/facebook/presto/features/binder/FeatureToggleTests.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.features.binder;
+
+import com.facebook.presto.features.annotations.FeatureToggle;
+import com.facebook.presto.features.classes.Feature01;
+import com.facebook.presto.features.classes.Feature01Impl01;
+import com.facebook.presto.features.classes.Feature01Impl02;
+import com.facebook.presto.features.config.FeatureConfiguration;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Guice;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.google.inject.Provider;
+import com.google.inject.TypeLiteral;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static com.facebook.presto.features.binder.FeatureToggleBinder.featureToggleBinder;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class FeatureToggleTests
+{
+    private final Map<String, FeatureConfiguration> map = new HashMap<>();
+
+    private static void sleep()
+    {
+        try {
+            Thread.sleep(6000);
+        }
+        catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Test
+    public void testHotReload()
```

Comment:
Shall we add more doc/comments for each test case to explain to end users when/how to use different types of injection?

Reply:
I will add comments to explain each use case